### PR TITLE
use default provider for fetch

### PR DESCRIPTION
**Pull request by thelostone-mc**
#### Description

`DEFAULT_PROVIDER` from chains.ts  is now used for fetch operations instead of relying on the provider configured in the user's wallet.
Any contract interaction would continue use the rpc url configured in the wallet.

Refs: https://github.com/dcgtc/dgrants/issues/509

### CODE REVIEW

File: app/src/views/GrantRegistryGrantDetail.vue [hunk 1]
```diff
@@ -352,12 +352,12 @@ function useGrantDetail() {
   const txHash = ref<string>();
 
   watch(
-    () => [grant.value, grantId.value, rounds.value, roundsCLRData.value, provider.value],
+    () => [grant.value, grantId.value, rounds.value, roundsCLRData.value, DEFAULT_PROVIDER],
```
Comment (gdixon) [inline app/src/views/GrantRegistryGrantDetail.vue:355]: `DEFAULT_PROVIDER` should always hold a value, so we shouldn't need to check it here 👍

File: app/src/views/GrantRegistryGrantDetail.vue [hunk 2]
```diff
@@ -352,12 +352,12 @@ function useGrantDetail() {
...
     () => {
       // enter loading state between loads
       loading.value = true;
       // ensure the computed props are ready before fetching data
-      if (donationToken.value && rounds.value && roundsCLRData.value && provider.value) {
+      if (donationToken.value && rounds.value && roundsCLRData.value && DEFAULT_PROVIDER) {
```
Comment (gdixon) [inline app/src/views/GrantRegistryGrantDetail.vue:360]: Or here

**Review by gdixon — APPROVED** [review]
Tiny nit, but all lgtm!! 🚀

File: app/src/components/TransactionStatus.vue [hunk 3]
```diff
@@ -50,7 +50,7 @@
 <script lang="ts">
 import { computed, defineComponent, Ref, ref, SetupContext, watch } from 'vue';
 import { getEtherscanUrl } from 'src/utils/utils';
-import useWalletStore from 'src/store/wallet';
+import { DEFAULT_PROVIDER } from 'src/utils/chains';
```
Comment (mds1) [inline app/src/components/TransactionStatus.vue:53]: I think I've found a few other instances we use the user's provider to read contract data, that we need to change to `DEFAULT_PROVIDER`. These link to lines on the last commit in this branch:

- L543 `cart.ts` :https://github.com/dcgtc/dgrants/blob/cc8f5485f8035b3a73e0eec34bea4ed351932734/app/src/store/cart.ts#L543
- L549 `cart.ts` :https://github.com/dcgtc/dgrants/blob/cc8f5485f8035b3a73e0eec34bea4ed351932734/app/src/store/cart.ts#L549
- L188 and L189 of `app/src/utils/utils`: https://github.com/dcgtc/dgrants/blob/cc8f5485f8035b3a73e0eec34bea4ed351932734/app/src/utils/utils.ts#L188-L189
- L464 of `grantRound.ts`: https://github.com/dcgtc/dgrants/blob/cc8f5485f8035b3a73e0eec34bea4ed351932734/app/src/utils/data/grantRounds.ts#L464
- L534 and 536 of `grantRound.ts`: https://github.com/dcgtc/dgrants/blob/cc8f5485f8035b3a73e0eec34bea4ed351932734/app/src/utils/data/grantRounds.ts#L534-L536

File: app/src/views/GrantRegistryGrantDetail.vue (hunk 1 above)
Comment (mds1) [reply app/src/views/GrantRegistryGrantDetail.vue:355]: Yep, and it's constant/not reactive/will never change, so we can remove it from the watcher entirely

**Review by mds1 — APPROVED** [review]

**Comment by mds1** [comment]
Just making sure @thelostone-mc has nothing else left to do on this PR before merging, so will let him handle it

**Pull request merged**